test/run_pass/test_net_http_outgoing_buffer.js

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2015-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var net = require('net');
+var http = require('http');
+var fs = require('fs');
+var assert = require('assert');
+
+var binaryMessage = fs.readFileSync('./resources/random_binarydata');
+assert.notEqual(binaryMessage.length, 0);
+
+var server = http.createServer(function(req, res) {
+  req.on('data', function(data) {
+    assert.equal(data.compare(binaryMessage), 0);
```

Comment:
A small problem with this is that you can not be sure how many parts the message is coming in.
In this case (as the data is small) it is expected that the body is received in one go. However if a bigger
data is sent you would need to wait for all of the contents, join them together and process the result.

ex,:
```
var chunks = [];
req.on('data', function(data) {
  chunks.push(data);
});
req.on('end', function(){
  var full_body = Buffer.concat(chunks);
  ...
});
```

Reply:
Could we put the comparison check in the `process.exit` callback? Because if the 'data' event is not triggered no checks will be done. Or maybe just put a simple "message received" check in the process exit and keep the assert check here.

Reply:
You are right
The reason I left it like this is we are sending a small amount of data 
However it's way better as you suggested to keep consistency and make the test scalable if needed later